node/pkg/watchers/evm/reobserve.go

```diff
@@ -0,0 +1,136 @@
+package evm
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"time"
+
+	"github.com/certusone/wormhole/node/pkg/watchers/evm/connectors"
+	eth_common "github.com/ethereum/go-ethereum/common"
+	"github.com/wormhole-foundation/wormhole/sdk/vaa"
+	"go.uber.org/zap"
+)
+
+// handleReobservationRequest performs a reobservation request and publishes any observed transactions.
+func (w *Watcher) handleReobservationRequest(ctx context.Context, chainId vaa.ChainID, txID []byte, ethConn connectors.Connector) (numObservations uint32, err error) {
+	// This can't happen unless there is a programming error - the caller
+	// is expected to send us only requests for our chainID.
+	if chainId != w.chainID {
+		return 0, fmt.Errorf("unexpected chain id: %v", chainId)
+	}
+
+	tx := eth_common.BytesToHash(txID)
+	w.logger.Info("received observation request", zap.String("tx_hash", tx.Hex()))
+
+	// SECURITY: Load the block number before requesting the transaction to avoid a
+	// race condition where requesting the tx succeeds and is then dropped due to a fork,
+	// but blockNumberU had already advanced beyond the required threshold.
+	//
+	// In the primary watcher flow, this is of no concern since we assume the node
+	// always sends the head before it sends the logs (implicit synchronization
+	// by relying on the same websocket connection).
+	blockNumberU := atomic.LoadUint64(&w.latestFinalizedBlockNumber)
```

Comment:
this does not seem same to assume that the watcher finalized block has any relation to the transaction from another connection. can `Reobserve` create a new watcher instance?

Reply:
I changed it to use the new `GetLatestBlocks` function to get the current finalized and safe blocks and pass them into this function rather than creating a new watcher. Creating a new watcher would be kind of messy since it would need a separate run function that didn't spin up all the runnables, etc.